src/libraries/PID/DNeutralShower_factory.cc

```diff
@@ -219,3 +312,55 @@ double DNeutralShower_factory::getFCALQuality( const DFCALShower* fcalShower, do
   return dFCALClassifier->GetMvaValue( mvaInputs );
 }
 
+int DNeutralShower_factory::check_TOF_match(DVector3 fcalpos, double rfTime, DVector3 vertex, vector< const DTOFPoint* > locTOFPoints, vector <float> &match_x, vector <float> &match_y) 
+{
+	
+  int global_tof_match = 0;
+  
+  for (vector< const DTOFPoint* >::const_iterator tof = locTOFPoints.begin(); tof != locTOFPoints.end(); tof++) {
+    
+    double xt = (*tof)->pos.X() - vertex.X();
+    double yt = (*tof)->pos.Y() - vertex.Y();
+    double zt = (*tof)->pos.Z() - vertex.Z();
+    double rt = sqrt(xt*xt + yt*yt + zt*zt);
+    double tt = (*tof)->t - (rt / TMath::C() * 1e7);
+    double dt = tt - rfTime;
+    xt *= fcalpos.Z() / zt;
+    yt *= fcalpos.Z() / zt;
+    double dx = fcalpos.X() - xt;
+    double dy = fcalpos.Y() - yt;
+    
+    if (fabs(dx) < TOF_FCAL_x_match_CUT && fabs(dy) < TOF_FCAL_y_match_CUT) {
+      if (fabs(dt) < TOF_RF_CUT) {
+	match_x.push_back(dx);
+	match_y.push_back(dy);
+	global_tof_match ++;
+      }
+    }
+  }
+
+  return global_tof_match;
+}
+
+int DNeutralShower_factory::check_SC_match(double phi, double rfTime, double SC_PHI_CUT, vector< const DSCHit* > locSCHits, vector <float> &match_phi) 
+{
+  int global_sc_match = 0;
+  
+  for (unsigned int i = 0; i < locSCHits.size(); i ++) {
+
+    const DSCHit *schits = locSCHits[i];
+    double t = schits->t;
+    double e = schits->dE;
+    double s = schits->sector;
+    double diff_t = t - rfTime;
+    double phi_sc = 6.0 * (s + 1) - 11.25 - 45.0 * TMath::DegToRad();
```

Comment:
this seems like an awful lot of hard-coded constants?  shouldn't the parameters be taken from the geometry, just like in DParticleID?